OVP/D3D9Client/VPlanet.cpp

```diff
@@ -1662,37 +1393,35 @@ void vPlanet::SetMicroTexture(LPDIRECT3DTEXTURE9 pSrc, int slot)
 }
 
 // ===========================================================================================
-//
-bool vPlanet::LoadMicroTextures()
+// static
+void vPlanet::LoadMicroTextures(LPDIRECT3DDEVICE9 pDev)
 {
-	LogOapi("Loading Micro Textures for %s", GetName());
-	char file[256];
-	for (int i=0; i<ARRAYSIZE(MicroCfg.Level); ++i) {
-		sprintf_s(file, 256, "Textures/%s", MicroCfg.Level[i].file);
-		if (D3DXCreateTextureFromFileA(GetDevice(), file, &MicroCfg.Level[i].pTex) == S_OK) {
-			D3DSURFACE_DESC desc;
-			if (MicroCfg.Level[i].pTex) {
-				MicroCfg.Level[i].pTex->GetLevelDesc(0, &desc);
-				MicroCfg.Level[i].size = double(desc.Width) / MicroCfg.Level[i].reso;
-				MicroCfg.Level[i].px = double(desc.Width);
-				DWORD mips = MicroCfg.Level[i].pTex->GetLevelCount();
-				LogOapi("Level %u, %s, %.1fpx/m, %.1fm, Mipmap count=%u", i, MicroCfg.Level[i].file, MicroCfg.Level[i].reso, MicroCfg.Level[i].size, mips);
+	for (auto &body : MicroCfgs)
+	{
+		for (auto &x : body.second.Level)
+		{
+			char file_path[MAX_PATH];
+			sprintf_s(file_path, MAX_PATH, "Textures/%s", x.file);
+			
+			// If texture is not loaded, load it
+			if (MicroTextures.find(x.file) == MicroTextures.end()) {
+				if (D3DXCreateTextureFromFileA(pDev, file_path, &x.pTex) == S_OK) {
+					LogAlw("Microtexture [%s] loaded", x.file);
+					MicroTextures[x.file] = x.pTex;
+				}
+				else {
+					LogErr("Failed to read microtexture [%s] for [%s]", file_path, body.first);
```

Comment:
LogErr("Failed to read microtexture [%s] for [%s]", file_path, body.first.**c_str()**);
would be correct as body.first is std::string...

Reply:
Thanks for the report. This should be fixed.